src/amqp.c

```diff
@@ -102,13 +102,18 @@ static char *amqp_bytes_to_new_string(amqp_bytes_t data)
 	return str;
 }
 
+/**
+ * Callback function to consume message envelope from AMQP queue.
+ *
+ * Returns true on success or false if the read callback is not set.
```

Comment:
This comment doesn't seem to be related to this callback. Shouldn't `on_receive` returns true always so we avoid that `l_io_set_read_handler` stop calling it?

Reply:
It is related to line 135.